Jenkinsfile

```diff
@@ -75,15 +75,19 @@ def setupNodeAndTest(version, filter='', testSuite='test') {
                 [ -s "$NVM_DIR/nvm.sh" ] && . "$NVM_DIR/nvm.sh"
                 nvm install ${version}
                 nvm use ${version}
-                npm install mocha-jenkins-reporter --save-dev
-                curl -O -u "\${ARTIFACTORY_USER}:\${ARTIFACTORY_PW}" "https://na.artifactory.swg-devops.com/artifactory/cloudant-sdks-maven-local/com/ibm/cloudant/${env.DBCOMPARE_NAME}/${env.DBCOMPARE_VERSION}/${env.DBCOMPARE_NAME}-${env.DBCOMPARE_VERSION}.zip"
-                unzip ${env.DBCOMPARE_NAME}-${env.DBCOMPARE_VERSION}.zip
-                set +x
-                export COUCH_BACKEND_URL="https://\${DB_USER}:${dbPassword}@\${SDKS_TEST_SERVER_HOST}"
-                export COUCH_URL="${(testSuite == 'toxytests/toxy') ? 'http://localhost:3000' : ((testSuite == 'test-iam') ? '${SDKS_TEST_SERVER_URL}' : '${COUCH_BACKEND_URL}')}"
-                set -x
-                ./node_modules/mocha/bin/mocha.js --reporter mocha-jenkins-reporter --reporter-options junit_report_path=./test/test-results.xml,junit_report_stack=true,junit_report_name=${testSuite} ${filter} ${testRun}
-              """
+                """
+              withNpmEnv('ARTIFACTORY_DOWN', registryArtifactoryDown) {
+                sh """
```

Comment:
Historically the NVM commands and the rest of the `sh` script had to be contained in the same `sh` step because the NVM setting didn't persist across multiple `sh` invocations. I can't see what version the tests are actally using in the build logs so we'll have to dig a bit deeper. I suspect this may need to be refactored to have the `withNpmEnv` enclose an entire single `sh` step, but I see that caused some other problems before.

Since it looked like the problem happened during install maybe we can resolve by doing the `nvm install` first and then `nvm use` within the same sh step as the test invocation. Otherwise we might need to check the nvm docs and see if we can flag off whatever it is doing.